File: cpp/containsDuplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using std::vector, 
    std::unordered_map, 
    std::cout, 
    std::endl;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> hash;
        for (auto num : nums) {
            if (hash.find(num) == hash.end()) {
                hash[num] = 1;
            } else {
                return true;
            }
        }
        return false;
    }
};


int main(int argc, char const *argv[]){
    Solution s;
    vector<int> nums = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    cout << s.containsDuplicate(nums) << endl;
    return 0;
}

File: python/containsDuplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        test = set()
        for num in nums:
            if num in test:
                return True
            else:
                test.add(num)
        return False
    
if __name__ == "__main__":
    so = Solution()
    print(so.containsDuplicate([1,2,3,4]))